daemon/logger/logger.go

```diff
@@ -88,7 +89,7 @@ type ReadConfig struct {
 // LogReader is the interface for reading log messages for loggers that support reading.
 type LogReader interface {
 	// ReadLogs reads logs from underlying logging backend.
-	ReadLogs(ReadConfig) *LogWatcher
+	ReadLogs(context.Context, ReadConfig) *LogWatcher
```

Comment:
Double checked if this interface change would affect logging plugins, but those implement a separate interface; https://github.com/moby/moby/blob/da3cc1c8eb2928bc159bdb1719ca20895beb9990/daemon/logger/plugin.go#L21-L27

And there's an adaptor for those; see further up the diff 😄